pallets/api/src/fungibles/mod.rs

```diff
@@ -159,14 +159,38 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Reads the state of the fungible asset based on the provided key.
+		///
+		/// This function matches the key to determine the type of state query and returns the
+		/// encoded result.
+		///
+		/// # Arguments
+		/// * `key` - An instance of `FungiblesKey<T>`, which specifies the type of state query and
+		/// 		  the associated parameters.
+		///
+		/// # Returns
+		/// A vector of bytes representing the encoded result of the state query.
+		pub fn read_state(key: FungiblesKey<T>) -> Vec<u8> {
```

Comment:
Still maintain we should avoid using Key and use something more abstract. Also, as this is defined in the fungibles pallet, I don't see that we need the Fungibles prefix. The pallet norm is Error and Event, not FungiblesError and FungiblesEvent. Suggest something like Query or Read.

Whilst perhaps not applicable to this pallet, I'm sure there will come a time where there is a query which doesn't take an argument or performs multiple reads to calculate a result.

Reply:
https://github.com/r0gue-io/pop-node/issues/127